exporters/secret/exporter.go

```diff
@@ -0,0 +1,162 @@
+package secret
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8srestconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	fieldSelector = "type=kubernetes.io/tls"
+	certType      = "secret"
+)
+
+type Config struct {
+	Namespaces []string
+}
+
+type Exporter struct {
+	cert      *prometheus.Desc
+	ctx       context.Context
+	k8sClient *kubernetes.Clientset
+	logger    micrologger.Logger
+
+	namespaces []string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Namespaces: []string{},
+	}
+}
+
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	e.logger.Log("info", "start collecting metrics")
+
+	// If no namespace whitelist is provided then we check all namespaces
+	if len(e.namespaces) == 0 {
```

Comment:
I think there is no need to iterate on the namespaces like this.
If you want to get secrets from all namespaces, you can simply list them on the "" namespace.
If you'll use the CtrlClient, you simply omit the Namespace Option and it looks to all namespaces by default